src/packages/cli/src/bin.ts

```diff
@@ -198,6 +209,51 @@ async function main(): Promise<number> {
   return 0
 }
 
+/**
+ * Get a unique identifier for the project by hashing
+ * the directory with `schema.prisma`
+ */
+async function getProjectHash(): Promise<string> {
+  const schemaPath = await getSchemaPath()
```

Comment:
So the schema could be somewhere else because we have the arg `--schema=/path/to/myschema.prisma` so this could return null, then you do a hash of null?

Reply:
Oh I was thinking you were using https://github.com/prisma/prisma/blob/902d4384fbadc9e27635c76e43f1bd4d25c0ba7e/src/packages/sdk/src/cli/getSchema.ts#L13 but you're using your own thing apparently

Reply:
Currently `getSchemaPath` will search for the `schema.prisma` file and default to the `cwd` if it can't find it.

As I read the comment I discovered that there's already a helper function which handles all cases: 
https://github.com/prisma/prisma/blob/c1624d9b953c8cf9396b3b7994f49a4fea5e9c95/src/packages/sdk/src/cli/getSchema.ts#L13 

Using it will require me to parse the `--schema` argument inside `bin.ts` and some commands don't have it. Since a project can exist prior to the schema existing (pre `prisma init` call) perhaps we should use the path of `package.json` as the path. 

Any thoughts?

Reply:
The parsing is done here https://github.com/prisma/prisma/blob/c1624d9b953c8cf9396b3b7994f49a4fea5e9c95/src/packages/cli/src/bin.ts#L57

Reply:
@2color 
> Since a project can exist prior to the schema existing (pre prisma init call) perhaps we should use the path of package.json as the path.
The package.json may not exists as well actually

Reply:
I changed this to use the `getSchemaPath` method from the SDK and defaulting to `cwd` if it can't be found. This is fine from the checkpoint-client's perspective as its cache is based on the CLI installation path.